Frameworks/Foundation/NSOperation.mm

```diff
@@ -304,16 +312,15 @@ - (void)waitUntilFinished {
 */
 - (NSArray*)dependencies {
     std::lock_guard<std::recursive_mutex> lock(_dependenciesLock);
-    NSArray* copy = [[_dependencies copy] autorelease];
-    return copy;
+    return [_dependencies allObjects];
 }
 
 /**
  @Status Interoperable
 */
 - (void)dealloc {
     _dependenciesLock.lock();
```

Comment:
>   _dependenciesLock.lock() [](start = 1, length = 27)

can we be consistent with the style of locking? I prefer lock_guard as its RAII so there is less chance of accidentally leaving things locked.